compiler/rustc_middle/src/mir/terminator.rs

```diff
@@ -510,7 +512,7 @@ mod helper {
         #[define_opaque(Successors)]
         pub fn successors_for_value(&self, value: u128) -> Successors<'_> {
             let target = self.target_for_value(value);
-            (&[]).into_iter().copied().chain(Some(target).into_iter().chain(None))
+            (&[]).into_iter().copied().chain(Some(target)).chain(None)
```

Comment:
Some of the original explicit `into_iter()` could still be avoided with `std::iter::chain`, plus `iter()` for slices:
```suggestion
            [].iter().copied().chain(chain(Some(target), None))
```